test/t/unit/test_unit_init_completion.py

```diff
@@ -32,3 +32,27 @@ def test_redirect(self, bash, redirect):
             bash, "%s " % redirect, cwd="shared/default"
         )
         assert all(x in completion for x in "foo bar".split())
+
+    def test_redirect_mock(self, bash):
+        assert_bash_exec(
+            bash,
+            "_cmd1() { local cur prev words cword;_init_completion; } && "
+            "complete -F _cmd1 cmd1",
+        )
```

Comment:
> Hmm,. I wonder what's the idea behind this, is it intended to serve as a definer of completion for `cmd1` that is used in `test_redirect_2` below? If so, I think it should be made a `@pytest.fixture`

Ah, yes. I hadn't the idea to use `@pytest.fixture`! Thank you for pointing it out. I have updated the test (d5845d7aaf2eba71c5c3fc06b47da2e4dccf7970..3bc4376113e75e4c50a6d58cb0fdb76a56c8780b).

> Or alternatively, it could be removed altogether because doing so doesn't make `test_redirect_2` fail, I guess because we end up loading `_minimal` for `cmd1` and testing with it.

I first tried it, but even if the redirection completion fails, the test doesn't fail because `_minimal` instead generates the filenames as arguments, which are originally expected as redirection words. I wanted a mock completion that does not produce any argument completions. Maybe we can use another command name that would never generate filenames as arguments, but in that case, the test depends on a specific completion, and also the intent might be less clear.